posthog/api/prompt.py

```diff
@@ -0,0 +1,103 @@
+from typing import Any, Dict, List, Union
+
+from dateutil import parser
+from rest_framework import exceptions, request, serializers, status, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.models.person.person import PersonDistinctId
+from posthog.models.prompt import PromptSequenceState, get_active_prompt_sequences
+
+
+class PromptSequenceStateSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = PromptSequenceState
+        fields = ["key", "last_updated_at", "step", "completed", "dismissed"]
+
+
+class PromptSequenceStateViewSet(StructuredViewSetMixin, viewsets.ViewSet):
+    """
+    Create, read, update and delete prompt sequences state for a person.
+    """
+
+    serializer_class = PromptSequenceStateSerializer
+
+    @action(methods=["PATCH"], detail=False)
+    def my_prompts(self, request: request.Request, **kwargs):
+        if not request.user.is_authenticated:  # for mypy otherwise check on distict_id below fails
+            raise exceptions.NotAuthenticated()
+        local_states: List[Dict[str, Any]] = []
+        for key in request.data:
+            parsed_state = dict(
+                request.data[key], last_updated_at=parser.isoparse(request.data[key]["last_updated_at"])
+            )
+            local_states.append(parsed_state)
+
+        my_prompts: Dict[str, Any] = {"sequences": [], "state": {}}
+        states_to_update: List[PromptSequenceState] = []
+        states_to_create: List[PromptSequenceState] = []
+
+        person_id = (
+            PersonDistinctId.objects.filter(distinct_id=request.user.distinct_id, team_id=self.team_id)
+            .values_list("person_id", flat=True)
+            .first()
+        )
+
+        if not person_id:
+            return Response("", status=status.HTTP_404_NOT_FOUND)
```

Comment:
This won't have a JSON payload, I believe we should raise a `exceptions.NotFound()` (not sure actually, since this is an `@action`, but we definitely should return the standard 404 JSON response)